src/dotnes.tests/RoslynTests.cs

```diff
@@ -412,4 +416,66 @@ public void Branch()
                 4C3485
                 """);
     }
+
+    [Fact]
+    public void PadInput()
+    {
+        AssertProgram(
+            csharpSource:
+                """
+                byte[] PALETTE = new byte[32] { 
+                    0x01,
+                    0x11,0x30,0x27,0x0,
+                    0x1c,0x20,0x2c,0x0,
+                    0x00,0x10,0x20,0x0,
+                    0x06,0x16,0x26,0x0,
+                    0x16,0x35,0x24,0x0,
+                    0x00,0x37,0x25,0x0,
+                    0x0d,0x2d,0x3a,0x0,
+                    0x0d,0x27,0x2a
+                };
+                pal_all(PALETTE);
+                ppu_on_all();
+                byte x = 40;
+                while (true) {
+                    if (pad_poll(0) != 0)
+                        x++;
+                    oam_spr(x, x, 0x10, 3, 0);
+                    ppu_wait_frame();
+                }
+                """,
+            expectedAssembly:
+                """
+                A9FA
+                A285
+                201182  ; JSR pal_all
+                208982  ; JSR ppu_on_all
+                A928
+                8D2403
+                A922
+                A286
+                A900
+                202686  ; JSR pad_poll
+                C900
+                F015
```

Comment:
This isn't right, I think `F015` should be a bit lower.

But this log message is generally right:
```
    Reading forward until IL_002c...
```
The IL looks like:

![image](https://github.com/user-attachments/assets/38cdbfda-79e0-4d31-bf47-95155d541bcf)